src/wallet/wallet.cpp

```diff
@@ -1380,6 +1418,7 @@ CPubKey CWallet::DeriveNewSeed(const CKey& key)
 
     // set the hd keypath to "s" -> Seed, refers the seed to itself
     metadata.hdKeypath     = "s";
```

Comment:
We need to keep `hdKeypath` forever as `CKeyMetadata` is backwards compatible.

Reply:
If hdKeypath is being kept forever, does it actually make sense to have the key_origin added to CKeyMetadata? You can calculate key_origin from hdKeypath using ParseHDKeypath() if needed, but everywhere that uses key_origin is converting it back to a string via WriteHdKeypath already as far as I can see? AddKeyOrigin is already filling out hdKeyPath so it would still achieve the point of the PR by the looks?

Reply:
`CWallet::GetKeyOrigin` uses the `key_origin` in `CKeyMetadata`. I strongly prefer storing key origin info in this form instead of having to parse strings every time key origin info is needed to build a PSBT.